draft-ietf-quic-transport.md

```diff
@@ -1449,20 +1417,121 @@ of three ways:
+
+
+A stateless reset is provided as an option of last resort for a server that does
+not have access to the state of a connection.  A server crash or outage might
+result in clients continuing to send data to a server that is unable to properly
+continue the connection.  A server that wishes to communicate a fatal connection
+error MUST use a CONNECTION_CLOSE frame if it has sufficient state to do so.
+
+To support this process, the server sends a stateless_reset_token value during
+the handshake in the transport parameters.  This value is protected by
+encryption, so only client and server know this value.
+
+A server that receives packets that it cannot process sends a packet in the
+following layout:
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+
+|0|C|K|  00001  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                     [Connection ID (64)]                      +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                                                               +
+|                                                               |
++                   Stateless Reset Token (128)                 +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                        Random Octets (*)                    ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+This packet uses the short header form with the shortest possible packet number
+encoding.  This minimizes the perceived gap between the last packet that the
+server sent and this packet.  The leading octet of the Stateless Reset Token
+will be interpreted as a packet number.  A server MAY use a different short
+header type, indicating a different packet number length, but this allows for
+the message to be identified as a stateless reset more easily using heuristics.
+
+A server copies the connection ID field from the packet that triggers the
+stateless reset.  A server omits the connection ID if explicitly configured to
+do so, or if the client packet did not include a connection ID.
+
+After the first short header octet and optional connection ID, the server
+includes the value of the Stateless Reset Token that it included in its
+transport parameters.
+
+After the Stateless Reset Token, the server pads the message with random
+octets.
+
+This design ensures that a stateless reset packet is - to the extent possible -
+indistinguishable from a regular packet.
+
```

Comment:
Perhaps add that the token MUST appear cryptographically strong random to an observer.